src/test/java/test/DynamicGraphTest.java

```diff
@@ -74,15 +73,19 @@ public void test8() {
     dg.addEdge(0, c1, x, y);
     assertFreeNodesEquals(dg, y, x);
     dg.setStatus( dg.getFreeNodes(), Status.RUNNING);
-
-    assertFreeNodesEquals(dg, a1, a2);
+    assertFreeNodesEquals(dg);
```

Comment:
In fact, the pull request changes the definition of a free dependent node: before a dependent node is free when its dependency is running, and after it is when its dependency is finished.

That's why `a1` and `a2` are not free.